### PR TITLE
:sparkles: Applications Inventory: Disable deletion of applications involved with ticket

### PR DESCRIPTION
**Pull request by gildub**
Resolves https://issues.redhat.com/browse/MTA-811

### CODE REVIEW

File: client/src/app/pages/applications/applications-table-assessment/applications-table-assessment.tsx [hunk 1]
```diff
@@ -445,9 +451,11 @@ export const ApplicationsTable: React.FC = () => {
     if (applicationWriteAccess) {
       actions.push({
         title: t("actions.delete"),
-        ...(row.migrationWave !== null && {
+        ...((row.migrationWave !== null || ticketByApplication(row.id)) && {
```
Comment (ibolton336) [inline client/src/app/pages/applications/applications-table-assessment/applications-table-assessment.tsx:454]: Is this fix exclusive to the assessments page?

**Review by ibolton336 — CHANGES_REQUESTED** [review]

File: client/src/app/pages/applications/applications-table-assessment/applications-table-assessment.tsx (hunk 1 above)
Comment (gildub) [reply client/src/app/pages/applications/applications-table-assessment/applications-table-assessment.tsx:454]: Yeah it's needed for analyze table too, I forgot, thanks

Comment (ibolton336) [reply client/src/app/pages/applications/applications-table-assessment/applications-table-assessment.tsx:454]: Np. Also after testing - The tooltip is showing up for the wrong application in the table. Something is wrong with the association logic for tickets <-> applications.

File: client/src/app/pages/applications/applications-table-analyze/applications-table-analyze.tsx [hunk 2]
```diff
@@ -218,6 +219,11 @@ export const ApplicationsTableAnalyze: React.FC = () => {
     onDeleteApplicationError
   );
 
+  const { tickets } = useFetchTickets();
+
+  const ticketByApplication = (id: number) =>
+    tickets.some((ticket) => ticket.id === id);
```
[the remark this answers is not in the document]
Comment (ibolton336) [reply client/src/app/pages/applications/applications-table-analyze/applications-table-analyze.tsx:225]: Checking the ticket ID against the app ID wont work. Need to check the ticket's associated application ID against the application ID when iterating through the apps in question.

**Review by ibolton336 — CHANGES_REQUESTED** [review]

**Pull request closed without merging**